Makefile

```diff
@@ -35,31 +35,24 @@ help: ## Show this help.
 ---------------- : ## ----------------
 
 .PHONY: setup
-setup: ## Set up a development environment
-ifeq ($(OS),Linux)
-	changes="`sudo snap install --no-wait codespell`"
-	changes="$$changes `sudo snap install --no-wait ruff`"
-	changes="$$changes `sudo snap install --no-wait shellcheck`"
-	changes="$$changes `sudo snap install --classic --no-wait astral-uv`"
-	for change in $$changes; do
-		snap watch $$change
-	done
-else ifeq ($(OS),Windows_NT)
-	pipx install uv
-	choco install shellcheck
-else ifeq ($(OS),Darwin)
-	brew install uv
-	brew install shellcheck
-endif
-ifneq ($(OS),Linux)
-	uv tool install codespell
-	uv tool install ruff
-	uv tool update-shell
-endif
+setup: install-uv setup-precommit ## Set up a development environment
+	uv sync --frozen --all-extras
+
```

Comment:
So I guess the workflow here is, first thing you do is run `make setup`, then after that you run your targets directly (like `make test-unit` or `make lint`) and those targets will install what they need.  I like it.

Reply:
Yeah, or if you're in CI you only need to run the relevant other `setup`